File: ruby-calculator/calculator.rb
# File:  calculator.rb

class Calculator

  def initialize() 
  end

  def add( input ) 
    if input.empty?
      return 0
    else
      #return Integer(input)
      return ("0," + input).split(',').inject{|l,r| Integer(l) + Integer(r)}
# {|l,r| Integer(l) + Integer(r)}
    end
  end

end

File: sbt-stringcalculator/src/main/scala/Calculator.scala

class Calculator{}

object Calculator {

  def apply(input: String) : Int = {
    sum(input)
  }

  var regexp = "([^,]*),(.*)".r

  def sum(input: String):Int ={
    input match {
      case "" => 0 
      case _ => input.split(",").foldLeft(0)(_+Integer.valueOf(_))
    }
  }
}
